webcam-capture-drivers/driver-ffmpeg-cli/src/main/java/com/github/sarxos/webcam/ds/ffmpegcli/FFmpegCliDevice.java

```diff
@@ -1,260 +1,218 @@
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public Dimension[] getResolutions() {
+        return resolutions;
+    }
+
+    @Override
+    public Dimension getResolution() {
+        if (resolution == null) {
+            resolution = getResolutions()[0];
+        }
+        return resolution;
+    }
+
+    private String getResolutionString() {
+        Dimension d = getResolution();
+        return String.format("%dx%d", d.width, d.height);
+    }
+
+    @Override
+    public void setResolution(Dimension resolution) {
+        this.resolution = resolution;
+    }
+
+    public void open() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                open.set(true);
+                try {
+                    startProcess();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            }
+        }).start();
+    }
+
+    @Override
+    public void close() {
+        if (!open.compareAndSet(true, false)) {
+            return;
+        }
+
+        process.destroy();
+
+        try {
+            process.waitFor();
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (disposed.compareAndSet(false, true) && open.get()) {
+            close();
+        }
+    }
+
+    @Override
+    public boolean isOpen() {
+        return open.get();
+    }
+
+    public String[] buildCommand() {
+        return new String[]{
+                "ffmpeg",
+                "-loglevel", "panic",          // suppress ffmpeg headers
+                "-f", "video4linux2",          // camera format
+                "-r", "50",                    // requested FPS
+                "-s", getResolutionString(),   // frame dimension
+                "-i", vfile.getAbsolutePath(), // input file
+                "-vcodec", "rawvideo",         // raw output
+                "-f", "rawvideo",              // raw output
+                "-vsync", "vfr",               // avoid frame duplication
+                "-pix_fmt", "bgr24",           // output format as bgr24
+                "-",                           // output to stdout
+        };
+    }
+
+    public BufferedImage getImage() {
+        if (!open.get()) {
+            return null;
+        }
+        return image;
+    }
+
+    @Override
+    public ByteBuffer getImageBytes() {
+        if (!open.get()) {
+            return null;
+        }
+
+        ByteBuffer byteBuffer = ByteBuffer.allocate(resolution.width * resolution.height * 3);
+        getImageBytes(byteBuffer);
+
+        return byteBuffer;
+    }
+
+    @Override
+    public void getImageBytes(ByteBuffer byteBuffer) {
+        DataBuffer dataBuffer = image.getRaster().getDataBuffer();
+
+        byte[] pixelData = ((DataBufferByte) dataBuffer).getData();
+        byteBuffer.wrap(pixelData);
```

Comment:
Are you sure you would like to return wrapped original buffer and not copy? Any changes in the underlying buffer will affect other threads reading the image data.